transport/tchannel/handler_test.go

```diff
@@ -663,3 +663,38 @@ func TestGetSystemError(t *testing.T) {
 		})
 	}
 }
+
```

Comment:
should we add an integration test where the client makes a call with a short timeout, and the handler blocks until timeout, and we verify that there's no unexpected log messages?

Reply:
If we're using yarpc for outbound and inbound integration test, the error would not be reproduced. Because the outbound yarpc would handle the context cancelling very well.
I used to try to build the integration test with that. However, the issue could not be reproduced.
In reality, the caller of our system is a python based service, where it does not handle the outbound timeout the same way like yarpc. That makes it hard to build integration test inside yarpc to verify this specific behavior.

However, the worst thing could happen is that the log still gets emitted even when the caller timed out. That's the same behavior we currently have. So it would not become worse.

Reply:
Sorry for the delay here @fangzhou37ub -- can we pass a test logger (e.g., http://go.uber.org/zap/zaptest/observer) to the server, and verify that there's no logs emitted?

Without the change, the test should have logs, it doesn't matter if the caller timeout expires as well, we should still see logs?